### PR TITLE
Adding new launchpad flow for VideoPress!

### PR DESCRIPTION
**Pull request by roundhill**
#### Proposed Changes

* Adds a new launchpad checklist for VideoPress/Videomaker onboarding:

Screen Shot 2022-09-16 at 11 33 39 AM

* Related: D87583-code.

#### Testing Instructions

* Apply D87583 to your sandbox.
* Start with http://calypso.localhost:3000/setup/intro?flow=videopress, and complete a signup for a site.
* You should end up at the launchpad. Note that the active step is `Upload your first video`.
* Upload a video to the site.
* Return to the launchpad, and the step should be marked as completed.
* Click `Launch Video Site`, it should work properly.

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

### CODE REVIEW

**Review by thedebian — CHANGES_REQUESTED** [review]
Code looks good.
When I uploaded a video, the launchpad task wasn't updated, so I couldn't go further and click on "launch site"
Capture d’écran 2022-09-22 à 10 25 17

File: client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/task-helper.ts [hunk 1]
```diff
@@ -91,6 +94,42 @@ export function getEnhancedTasks(
 						},
 					};
 					break;
+				case 'videopress_setup':
+					taskData = {
+						title: translate( 'Set up your Video Site' ),
```
Comment (jgcaruso) [inline client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/task-helper.ts]:
```suggestion
						title: translate( 'Set up your Video site' ),
```
Not entirely sure about this, but to match the casing of the initial launch page, `site` is lowercased there as well.

File: client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/task-helper.ts [hunk 2]
```diff
@@ -91,6 +94,42 @@ export function getEnhancedTasks(
...
+				case 'videopress_upload':
+					taskData = {
+						title: translate( 'Upload your first video' ),
+						actionUrl: `/site-editor/${ siteSlug }`,
+						isCompleted: videoPressUploadCompleted,
+					};
+					break;
+				case 'videopress_launched':
+					taskData = {
+						title: translate( 'Launch Video Site' ),
```
Comment (jgcaruso) [inline client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/task-helper.ts]:
```suggestion
						title: translate( 'Launch Video site' ),
```
Another "not entirely sure about" comment, even though it looks odd, should `site` be lowercased?

File: client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/translations.tsx [hunk 3]
```diff
@@ -18,6 +18,10 @@ export function getLaunchpadTranslations( flow: string | null ): TranslatedLaunc
 			translatedStrings.flowName = translate( 'Link in Bio' );
 			translatedStrings.sidebarTitle = translate( 'Your Link in Bio is ready to launch!' );
 			break;
+		case VIDEOPRESS_FLOW:
+			translatedStrings.flowName = translate( 'Video' );
+			translatedStrings.sidebarTitle = translate( 'Your Video Site is ready to launch!' );
```
Comment (jgcaruso) [inline client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/translations.tsx]:
```suggestion
			translatedStrings.sidebarTitle = translate( 'Your Video site is ready to launch!' );
```
here also?

**Review by jgcaruso — APPROVED** [review]
Tested alongside D87583-code and it worked as expected!

I could be completely wrong about the casing of `site` in the various actions, so those comments are not blocking at all.

**Pull request merged**